analyzers/tests/SonarAnalyzer.UnitTest/Rules/TimestampsShouldNotBeUsedAsPrimaryKeys.cs

```diff
@@ -0,0 +1,73 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using SonarAnalyzer.Rules.CSharp;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+[TestClass]
+public class TimestampsShouldNotBeUsedAsPrimaryKeysTest
+{
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_CSharp() =>
+        CreateVerifier<TimestampsShouldNotBeUsedAsPrimaryKeys>("TimestampsShouldNotBeUsedAsPrimaryKeys.CSharp.cs").Verify();
+
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_CSharp9() =>
+        CreateVerifier<TimestampsShouldNotBeUsedAsPrimaryKeys>("TimestampsShouldNotBeUsedAsPrimaryKeys.CSharp9.cs")
+            .WithOptions(ParseOptionsHelper.FromCSharp9)
+            .Verify();
+
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_NoReferenceToEntityFramework_CSharp() =>
+        new VerifierBuilder<TimestampsShouldNotBeUsedAsPrimaryKeys>()
+            .AddPaths("TimestampsShouldNotBeUsedAsPrimaryKeys.NoReferenceToEntityFramework.CSharp.cs")
+            .Verify();
+
+#if NET
+    [TestMethod]
+    public void TimestampsShouldNotBeUsedAsPrimaryKeys_EntityFrameworkCore_CSharp() =>
+        CreateVerifier<TimestampsShouldNotBeUsedAsPrimaryKeys>("TimestampsShouldNotBeUsedAsPrimaryKeys.EntityFrameworkCore.CSharp.cs").Verify();
+#endif
+
+    private static VerifierBuilder CreateVerifier<TAnalyzer>(string testFileName)
+        where TAnalyzer : DiagnosticAnalyzer, new()
+    {
+        var builder = new VerifierBuilder<TAnalyzer>()
+            .AddReferences(NuGetMetadataReference.SystemComponentModelAnnotations())
+
+#if NET
+            .AddReferences(NuGetMetadataReference.MicrosoftEntityFrameworkCore("7.0.0"))
+            .AddReferences(NuGetMetadataReference.MicrosoftEntityFrameworkCoreAbstractions("7.0.0"));
+        return AddTestFileWithExtraLinePrepended(builder, testFileName, "using Microsoft.EntityFrameworkCore;");
+#else
+            .AddReferences(NuGetMetadataReference.MicrosoftEntityFramework("6.0.0"));
+        return AddTestFileWithExtraLinePrepended(builder, testFileName, "using System.Data.Entity;using ModelBuilder = System.Data.Entity.DbModelBuilder;");
+#endif
+
+    }
+
+    private static VerifierBuilder AddTestFileWithExtraLinePrepended(VerifierBuilder builder, string testFileName, string extraLine)
+    {
+        var content = File.ReadAllText($@"TestCases\{testFileName}");
+        return builder.AddSnippet($"{extraLine}{Environment.NewLine}{content}", testFileName);
```

Comment:
When you run tests on files that are added from the disk, there is a logic that will create a second file so if concurrency is enabled for the rule it can be properly tested (For concurrency to kick in at least 2 syntax trees are needed). Now that you are using the AddSnippet API this is not happening.
My suggestion is two simply duplicate the test files for now for .NET and .NET Framework and open an issue to come up with a proper solution. This logic might come handy for other rules to, so I think it is better to implement it in the builder itself.

Reply:
I managed to extract the code which differs between EF6 and EF Core. Because it's a FN, I didn't feel the need to run it on both frameworks. I moved it to a separate file and only added EF Core references to it.